src/domain/community/invitation/invitation.service.ts

```diff
@@ -0,0 +1,162 @@
+    @Inject(WINSTON_MODULE_NEST_PROVIDER) private readonly logger: LoggerService
+  ) {}
+
+  async createInvitation(
+    invitationData: CreateInvitationInput
+  ): Promise<IInvitation> {
+    const invitation: IInvitation = Invitation.create(invitationData);
+
+    invitation.authorization = new AuthorizationPolicy();
+    // save the user to get the id assigned
+    await this.invitationRepository.save(invitation);
+
+    invitation.lifecycle = await this.lifecycleService.createLifecycle(
+      invitation.id,
+      invitationLifecycleConfig
+    );
+
+    return await this.invitationRepository.save(invitation);
+  }
+
+  async deleteInvitation(
+    deleteData: DeleteInvitationInput
+  ): Promise<IInvitation> {
+    const invitationID = deleteData.ID;
+    const invitation = await this.getInvitationOrFail(invitationID);
+
+    if (invitation.authorization)
+      await this.authorizationPolicyService.delete(invitation.authorization);
+
+    const result = await this.invitationRepository.remove(
+      invitation as Invitation
+    );
+    result.id = invitationID;
+    return result;
+  }
+
+  async getInvitationOrFail(
+    invitationId: string,
+    options?: FindOneOptions<Invitation>
+  ): Promise<Invitation | never> {
+    const invitation = await this.invitationRepository.findOne({
+      ...options,
+      where: {
+        ...options?.where,
+        id: invitationId,
+      },
+    });
+    if (!invitation)
+      throw new EntityNotFoundException(
+        `Invitation with ID ${invitationId} can not be found!`,
+        LogContext.COMMUNITY
+      );
+    return invitation;
+  }
+
+  async save(invitation: IInvitation): Promise<IInvitation> {
+    return await this.invitationRepository.save(invitation);
+  }
+
+  async getInvitationState(invitationID: string): Promise<string> {
+    const invitation = await this.getInvitationOrFail(invitationID);
+    const lifecycle = invitation.lifecycle;
+    if (lifecycle) {
+      return await this.lifecycleService.getState(lifecycle);
+    }
+    return '';
+  }
+
+  async getInvitedUser(invitation: IInvitation): Promise<IUser> {
+    const user = await this.userService.getUserOrFail(invitation.invitedUser);
+    if (!user)
+      throw new RelationshipNotFoundException(
+        `Unable to load User for invitation ${invitation.id} `,
+        LogContext.COMMUNITY
+      );
+    return user;
+  }
+
+  async getInvitedBy(invitation: IInvitation): Promise<IUser> {
+    const user = await this.userService.getUserOrFail(invitation.invitedBy);
+    if (!user)
+      throw new RelationshipNotFoundException(
+        `Unable to load User that created invitation ${invitation.id} `,
+        LogContext.COMMUNITY
+      );
+    return user;
+  }
+
+  async findExistingInvitations(
+    userID: string,
+    communityID: string
+  ): Promise<IInvitation[]> {
+    const existingInvitations = await this.invitationRepository
+      .createQueryBuilder('invitation')
```

Comment:
this is pretty straightforward query, can be done with `find` + options

Reply:
I did it for the other one, but did not see how to do a find on a child entry, if you know how please go ahead and update the code...